tests/unit/test_pack_unpack.py

```diff
@@ -0,0 +1,75 @@
+from collections import OrderedDict
+
+import pytest
+from authlib.common.encoding import json_dumps
+
+from didcomm.common.resolvers import ResolversConfig
+from didcomm.common.types import VerificationMethodType, VerificationMaterial, VerificationMaterialFormat
+from didcomm.message import MessageOptionalHeaders, MessageRequiredHeaders, MessageBody, Message
+from didcomm.pack_signed import pack_signed
+from didcomm.secrets.secrets_resolver import Secret
+from tests.common.example_resolvers import ExampleSecretsResolver, ExampleDIDResolver
+
+
+@pytest.mark.asyncio
+async def test_pack_signed():
+    alice_did = "did:example:alice"
+
+    alice_secret = Secret(
+        kid="did:example:alice#key-1",
+        type=VerificationMethodType.ED25519_VERIFICATION_KEY_2018,
+        verification_material=VerificationMaterial(
+            format=VerificationMaterialFormat.JWK,
+            value=json_dumps({
+                "kty": "OKP",
+                "d": "uuAE6HmqnCnVjkF0ygjZMQiHeYIvI3Qcwh_2SjGMG-o",
+                "crv": "Ed25519",
+                "x": "z0x6oKBZ-ehwn_tkBzbhav132eQ7vmj5s5Xen00rtW0"
+            })
+        )
+    )
+
+    secrets_resolver = ExampleSecretsResolver([alice_secret])
+
+    message = Message(
+        id="1234567890",
+        typ="application/didcomm-plain+json",
+        type="http://example.com/protocols/lets_do_lunch/1.0/proposal",
+        frm="did:example:alice",
+        to=[
+            "did:example:bob",
+            "did:example:charlie"
+        ],
+        created_time=1516269022,
+        expires_time=1516385931,
+        body=OrderedDict({
+            "messagespecificattribute": "and its value"
+        })
+    )
+
+    resolvers_config = ResolversConfig(secrets_resolver, ExampleDIDResolver())
+
+    result = await pack_signed(message, alice_did, resolvers_config)
+
+    expected_packed_msg = json_dumps(OrderedDict({
+        "payload": "eyJpZCI6IjEyMzQ1Njc4OTAiLCJ0eXAiOiJhcHBsaWNhdGlvbi9kaWRjb21tLXBsYWluK2pzb24iLCJ0" +
```

Comment:
Not sure about this, but maybe we can put expected and input JSONs into files (white spaces need to be trimmed when loading it).